notebooks/Analyze_user_behavior_across_different_lending_protocols.ipynb

```diff
@@ -625,15 +881,56 @@
+      "text/plain": [
+       "<Figure size 1000x800 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
    "source": [
     "# Function to plot Venn diagrams for USD amounts\n",
     "def plot_venn_diagram_usd(amounts_dict, title):\n",
```

Comment:
I have no idea what's going on when you create `subsets`. That part is not readable at all. Pls split it to more rows and name the variable well.

Reply:
Also, I think the computations are wrong. You don't have enough information in `amounts_dict` to tell how much of the tokens are deposited e.g. both at zkLend and Nostra Mainnet. You have to compute that from the per-user data, just like the amount of users that deposit at both protocols.